### PR TITLE
Fix/use range slider ts errors

### LINKED ISSUE #1640: useRangeSlider example not updated
### Subject

useRangeSlider

### Description

As stated in the chakra-ui issue [#7955](https://github.com/chakra-ui/chakra-ui/issues/7955), the example in the docs is not updated.

### PR DESCRIPTION
**Pull request by DavideFrancescon**
Closes #1640 (definitively) 

## 📝 Description

As stated in the comment of chakra-ui [#7955](https://github.com/chakra-ui/chakra-ui/issues/7955) by @hooch, the pr #1641, did not include all the changes and fixes required, so with this I'm adding the remaining ones.

## ⛳️ Current behavior (updates)

`= useRangeSlider({min, max, defaultValue, rest})`

`<Actions actions={actions} min={min} max={max} stepToIndex={stepToIndex} stepToNumber={stepToNumber} />`

## 🚀 New behavior

`= useRangeSlider({min, max, defaultValue, ...rest})`

` <Actions
              actions={actions}
              min={defaultValue[0]}
              max={defaultValue[1]}
              stepToIndex={stepToIndex}
              stepToNumber={stepToNumber}
            />`

## 💣 Is this a breaking change (Yes/No):

No

## 📝 Additional Information

Resolved some TS typing issues with the Thumb component Props

### CODE REVIEW

File: configs/sandpack-contents/hooks/use-range-slider.js [hunk 1]
```diff
@@ -24,107 +33,136 @@ export default function App({ min, max, stepToNumber, stepToIndex, stepByNumber,
...
+            defaultValue={[25, 75]}
+            aria-label={['min', 'max']}
+          />
         </ChakraProvider>
-    </React.StrictMode>
-);`,
+      </React.StrictMode>,
+    )    
+  `,
     Instructions: `import { Box, Text, Kbd } from "@chakra-ui/react";
```
Comment (hooch) [inline configs/sandpack-contents/hooks/use-range-slider.js:167]: The `Instructions` component has a double-nesting of `<Text>` which generates `<p><p>..</p></p>`.
Suggest changing the outer usage to `<Text as='div'>` at L137.

File: configs/sandpack-contents/hooks/use-range-slider.js [hunk 2]
```diff
@@ -204,38 +242,57 @@ const ThumbIndexIcon = ({ bgColor }: Props) => {
...
+    
+    type Props = {
+      value: number
+      thumbIndex: number
+      thumbProps: any
+      bgColor: string
+      onKeyDownStepBy: (
+        e: React.KeyboardEvent<HTMLDivElement>,
+        thumbIndex: number,
+      ) => void
```
Comment (hooch) [inline configs/sandpack-contents/hooks/use-range-slider.js:261]: 👍

**Review by hooch — CHANGES_REQUESTED** [review]

File: configs/sandpack-contents/hooks/use-range-slider.js [hunk 3]
```diff
@@ -1,21 +1,30 @@
...
-};
-
-export default function App({ min, max, stepToNumber, stepToIndex, stepByNumber, defaultValue, ...rest}: Props) {
-    const {
+    App: `import { Badge, Box, chakra, Flex, useRangeSlider } from '@chakra-ui/react'
+    import Actions from './Actions'
+    import Instructions from './Instructions'
+    import Thumb, { onKeyDownStepByType } from './Thumb'
+    import { DOMElement } from 'react'
+
```
Comment (hooch) [inline configs/sandpack-contents/hooks/use-range-slider.js:7]: The two type imports after this one appear to be redundant as they generate linter warnings.

`import Thumb from "./Thumb"`

**Review by hooch — CHANGES_REQUESTED** [review]

File: configs/sandpack-contents/hooks/use-range-slider.js (hunk 3 above)
Comment (DavideFrancescon) [reply configs/sandpack-contents/hooks/use-range-slider.js:7]: oh shoot i deleted that but i think i forgot to add the changes before merging ahah thanks for noticing it

**Comment by hooch** [comment]
Looks good

**Review by hooch — CHANGES_REQUESTED** [review]

File: configs/sandpack-contents/hooks/use-range-slider.js [hunk 4]
```diff
@@ -24,107 +32,137 @@ export default function App({ min, max, stepToNumber, stepToIndex, stepByNumber,
...
+          index: 1,
         })
-
-    const markers = Array.from({ length: 3 }, (_, i) => i + 1).map((i) =>
-        getMarkerProps({ value: i * 25 })
-    );
-
-    const onKeyDownStepBy = (e, thumbIndex) => {
+    
+      const markers = Array.from({ length: 3 }, (_, i) => i + 1).((i) =>
```
[the remark this answers is not in the document]
Comment (DavideFrancescon) [reply configs/sandpack-contents/hooks/use-range-slider.js:47]: oh, nice catch, i don't know where it went missing but thanks for noticing. This pr started from a 2 rows changes and ended up as an almost complete redesign of the example ahah

**Pull request merged**